app/lib/cashu/types.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Tags are part of the data in a NUT-10 secret and hold additional data committed to
+ * and can be used for feature extensions.
+ *
+ * Tags are arrays with two or more strings being `["key", "value1", "value2", ...]`.
+ *
+Supported tags are:
+
+ * - `sigflag`: <str> determines whether outputs have to be signed as well
+ * - `n_sigs`: <int> specifies the minimum number of valid signatures expected
+ * - `pubkeys`: <hex_str> are additional public keys that can provide signatures (allows multiple entries)
+ * - `locktime`: <int> is the Unix timestamp of when the lock expires
+ * - `refund`: <hex_str> are optional refund public keys that can exclusively spend after locktime (allows multiple entries)
+ *
+ * @example
+ * ```typescript
+ * const tag: NUT10SecretTag = ["sigflag", "SIG_INPUTS"];
+ * ```
+ */
+export type NUT10SecretTag = [string, ...string[]];
+
+/**
+ * CAUTION: If the mint does not support spending conditions or a specific kind
+ * of spending condition, proofs may be treated as a regular anyone-can-spend tokens.
+ * Applications need to make sure to check whether the mint supports a specific kind of
+ * spending condition by checking the mint's info endpoint.
+ */
+export const WELL_KNOWN_SECRET_KINDS = ['P2PK'] as const;
+
+/**
+ * the kind of the spending condition
+ */
+export type WellKnownSecretKind = (typeof WELL_KNOWN_SECRET_KINDS)[number];
+
+/**
+ * The data from a parsed stringified NUT-10 secret
+ */
+export type NUT10SecretData = {
+  nonce: string;
+  data: string;
+  tags?: NUT10SecretTag[];
```

Comment:
I left out adding docs to the keys in this type. The types are repeated in `NUT10Secret`, so I'm not sure if there's a better way to do this or if I should just duplicate the docs.

`NUT10Secret` is the data format normal consumers of this lib should use. My thinking for having the `Nut10SecretData` and `ParsedNUT10Secret` be their own types is that we might want to construct our own NUT10Secrets rather than letting cashu-ts do that. [This](https://github.com/cashubtc/cashu-ts/pull/218) PR might change things because it will add some functionality to cash-ts for custom spending conditions

Reply:
I am not sure I understand why we have both NUT10Secret and NUT10SecretData. If we do need to have one way of sharing the common stuff would be to use inheritance or ts Pick/Omit utils